components/faq.tsx

```diff
@@ -54,13 +58,23 @@ export default function FaqPage({ fetchedFaqs }: { fetchedFaqs: AnsweredQuestion
           <div className="flex flex-row items-center gap-x-2">
             <button
               onClick={() => {
-                expandAll();
+                if (disclosuresStatus.every((status) => status)) {
+                  closeAll();
+                } else {
+                  expandAll();
+                }
```

Comment:
Extract into named function instead of anonymous lambda function

Reply:
done, made it FaqExpandCloseButton component